### PR TITLE
Rename NeedsDeprovisioning() method

### PR DESCRIPTION
**Pull request by zaneb**
Since 3452e801c9704bfbf07351c36925fb74b1039ea5 this method is no
longer well-named, since it works only under assumptions about the state
the Host is in when it is called. Rename it and make it a private method
of the HostStateMachine.

### CODE REVIEW

File: pkg/controller/baremetalhost/host_state_machine.go [hunk 1]
```diff
@@ -285,7 +301,7 @@ func (hsm *hostStateMachine) handleProvisioningError(info *reconcileInfo) action
 }
 
 func (hsm *hostStateMachine) handleProvisioned(info *reconcileInfo) actionResult {
-	if hsm.Host.NeedsDeprovisioning() {
+	if hsm.provisioningCancelled() {
```
Comment (dhellmann) [inline pkg/controller/baremetalhost/host_state_machine.go]: Would a host that has already completed provisioning end up here?

Comment (zaneb) [reply pkg/controller/baremetalhost/host_state_machine.go]: Yes. It's not a perfect name.

**Comment by maelk** [comment]
The previous integration test run was giving 404. Re-running to check the current state

**Comment by dhellmann** [comment]
Do we still want this change?

**Comment by zaneb** [comment]
> Do we still want this change?

It still feels like an improvement.

**Review by dhellmann — COMMENTED** [review]
/lgtm

**Comment by dhellmann** [comment]
I merged it by hand.

**Pull request merged**